bokehjs/src/coffee/source/ajax_data_source.coffee

```diff
@@ -16,20 +16,27 @@ define [
 
     setup : (plot_view, glyph) =>
       @pv = plot_view
-      @update()
+      @get_data(true)
       if @get('polling_interval')
-        @interval = setInterval(@update, @get('polling_interval'))
+        @interval = setInterval( @get_data, @get('polling_interval'), @get('overwrite'), @get('max_size'), @get('if_modified'))
 
-    update : () =>
+    get_data : (overwrite, max_size=0, if_modified=false) =>
       $.ajax(
         dataType: 'json'
+        ifModified: if_modified
         url : @get('data_url')
         xhrField :
           withCredentials : true
         method : @get('method')
         contentType : 'application/json'
       ).done((data) =>
-        @set('data', data)
+        if overwrite == true
```

Comment:
@bryevdv Using debugger, it seems that mode at this point is undefined, how possible?